code/edgemedia/src/main/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventHelper.kt

```diff
@@ -159,19 +149,29 @@ internal class MediaXDMEventHelper {
 
             // Append standard metadata to advertisingDetails
             metadata.forEach { (key, value) ->
-                run {
-                    if (!standardAdMetadataSet.contains(key)) {
-                        return@forEach
-                    }
-
-                    when (key) {
-                        MediaConstants.AdMetadataKeys.ADVERTISER -> advertisingDetails.advertiser = value
-                        MediaConstants.AdMetadataKeys.CAMPAIGN_ID -> advertisingDetails.campaignID = value
-                        MediaConstants.AdMetadataKeys.CREATIVE_ID -> advertisingDetails.creativeID = value
-                        MediaConstants.AdMetadataKeys.CREATIVE_URL -> advertisingDetails.creativeURL = value
-                        MediaConstants.AdMetadataKeys.PLACEMENT_ID -> advertisingDetails.placementID = value
-                        MediaConstants.AdMetadataKeys.SITE_ID -> advertisingDetails.siteID = value
-                    }
+                if (!standardAdMetadataSet.contains(key)) {
```

Comment:
Same as previous comment, as `when` only assigns the value for the defined keys, is this check needed?